### DISCUSSION TITLE
Clean-up naming of lifecycle hooks

### DISCUSSION THREAD

**Discussion by sschuberth** [kotest/kotest#4063] [6.0 changes]
I'd be great if we could clean-up / clarify the IMO very confusing naming of [lifecycle hooks](https://kotest.io/docs/framework/lifecycle-hooks.html) in 6.0. Quoting @sksamuel from [Slack](https://kotlinlang.slack.com/archives/CT0G9SD7Z/p1638114742201200?thread_ts=1638093985.199900&cid=CT0G9SD7Z):

> Yes the naming is shit

😄 

At the example of `after*`:

- Remove the current `afterTest` in favor of keeping only `afterAny`
- Rename the current `afterEach` to `afterTest` (yes, this could be problematic as `afterTest` would be doing something else than before now)

However, in general I feel that lifecycle suffixes should be aligned with test type names. So if test types are [removed](https://github.com/kotest/kotest/discussions/4055), or maybe renamed partly, this shoul dbe reflected in the lifecycle names. As an example, if `TestType.Test` was renamed to `TestType.Leaf`, also `afterLeaf` instead of `afterTest` should be used.

**Comment by sksamuel** [comment]
I'm on board with cleaning up the names.

**Reply by sschuberth** [reply]
How would we practically do this? Is there a version 6 branch this breaking change could be integrated to without merging the changes to `master` just now? (Which would prevent us from doing version 5 releases from `master` from then on.)

**Comment by sksamuel** [comment]
Version 5 releases can take place from the branch. Master is 6 development.

**Reply by sschuberth** [reply]
Milestone [6.0.0.M1](https://github.com/kotest/kotest/releases/tag/v6.0.0.M1) arrived quicker than I thought 😓 From skimming over the changes, I believe this renaming did not make it into it? Is it too late to do so now?

**Reply by sksamuel** [reply]
No, M1 is really just because some people have asked for a "stable" 6.0 build that fixes incompatibilities with kotlin 2.
I'd say 6.0 is still a month or two away. I need to fix all the plugin stuff so that we can run individual tests in JS/Native still.

**Reply by sschuberth** [reply]
So, what's the current status of renaming lifecycle hook? Has work already been done in this area? If not, do we still want this for 6.0 final?

**Reply by Kantis** [reply]
I think it's a bit late to add deprecations at this point? Perhaps we can plan for a 7.0 release for the start of 2026 which fixes this (and possibly other API cleanup?), and add the deprecations in 6.0?
